media/js/app/taxonomy/taxonomy.js

```diff
@@ -442,31 +448,45 @@
 
             // split the url.
             var urls = value.split(',');
+            var flagArray = [];
```

Comment:
Rather than messing around with this flagArray, I'd opt for doing the regex check twice -- once for the valid url check at line 462 and once for the getTheOnomy call at line 477.  Simplifies the code.